keps/sig-node/2371-cri-pod-container-stats/kep.yaml

```diff
@@ -16,10 +16,11 @@ approvers:
 creation-date: 2021-01-27
 last-updated: 2022-06-16
 status: implementable
-stage: alpha
-latest-milestone: "v1.26"
+stage: beta
```

Comment:
beta graduation says:

"cAdvisor stats provider may be marked as deprecated (depending on stability of new CRI based implementations)."

How will this be defined?

Also, for the performance validation - will those be ad-hoc or continuous tests?

I am also not clear on Windows graduation criteria. Earlier it says that Alpha will not look at Windows and Beta will. Should Windows be added to the graduation criteria for beta?

Reply:
which defined? stability? Honestly, I would consider "stability" a qualifier for beta anyway. I was thinking within the performance target and a full metric replacement. I've reworeded the statement.

for windows: I think windows stats already exist, but I am calling it out explicitly in beta criteria cc @jsturtevant @marosset

Reply:
cri changes are in place with https://github.com/kubernetes/kubernetes/pull/110754 and containerd changes in flight here https://github.com/containerd/containerd/pull/7099. 

Will need to update kubelet to read those fields for Windows

Reply:
The Windows support for the "/metrics/cadvisor" endpoint is not being worked on currently that I know of.  There are fewer stats at that endpoint for Windows but there are a some that need to added to containerd: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cadvisor/cadvisor_windows.go

Reply:
do you think targeting 1.27 is realistic for those changes @jsturtevant ?

Reply:
I am not sure for the `metrics/cadvisor changes`. I haven't looked into the scope of those changes.

If I could get a few more eyes on https://github.com/containerd/containerd/pull/7099 that would help move that along.  The kubelet changes should be minor since most of the changes have been made for Linux.

Reply:
is this resolved? Will we attempt to get this KEP for 1.27 with the goal of supporting Windows as well? Or moving to the next release?

Next release doesn't mean we cannot work on this feature this release by improving things like performance mentioned in graduation requirements

Reply:
fyi, @mansikulkarni96 opened: https://github.com/kubernetes/kubernetes/pull/116968 which we will target to get in for 1.28

Reply:
the containerd changes were included in contianerd 1.7